File: 253. Meeting Rooms II.py
class Solution:
    def minMeetingRooms(self, intervals: List[List[int]]) -> int:
        intervals.sort( key = lambda x : x[0] )
        rooms = []
        heapq.heapify( rooms )
        res = 0
        for interval in intervals:
            if len( rooms ) == 0:
                heapq.heappush( rooms, interval[1] )
            elif interval[0] >= rooms[0]:
                heapq.heappop( rooms )
                heapq.heappush( rooms, interval[1] )
            else:
                heapq.heappush( rooms, interval[1] )
        return len( rooms )

File: 253. Meeting Rooms II.java
class Solution {
    public int minMeetingRooms(int[][] intervals) {
        Arrays.sort( intervals, new Comparator<int[]>(){
           public int compare( int[] o1, int[] o2) {
               return o1[0] - o2[0];
           }
        });
        PriorityQueue<Integer> result = new PriorityQueue();
        for ( int[] interval : intervals ) {
            if ( result.size() == 0 ) {
                result.add( interval[1] );
            } else if( result.peek() <= interval[0] ) {
                result.poll();
                result.add( interval[1] );
            } else {
                result.add( interval[1] );
            }
        }
        return result.size();
    }
}